### PR TITLE
MVP of the debug cart panel

### PR DESCRIPTION
**Pull request by pjedrzejewski**
| Q               | A
| --------------- | -----
| Branch?         |  master
| Bug fix?        | no
| New feature?    | yes
| BC breaks?      | no
| Deprecations?   | no
| License         | MIT

Why? Sometimes, you want to debug what was inside of the cart on a specific request. While on shop pages you will see the cart summary widget, you do not know the details (id, ids and codes of products/variants), etc. This is useful when debugging promotions, taxes, etc. This is an MVP done during a hackathon, but can be extended with more data and details, especially about $.

Screenshot 2019-09-21 at 16 54 08
Screenshot 2019-09-21 at 16 54 11
Sylius___Your_shopping_cart

### CODE REVIEW

File: src/Sylius/Bundle/CoreBundle/Collector/DebugCartCollector.php [hunk 1]
```diff
@@ -0,0 +1,132 @@
...
+                'variantId' => $variant->getId(),
+                'variantCode' => $variant->getCode(),
+                'quantity' => $item->getQuantity(),
+                'productName' => $product->getName(),
+                'productId' => $product->getId(),
+            ];
+        }
+
+        $this->data = [
+            'cart_id' => $cart->getId(),
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Collector/DebugCartCollector.php]: Can we add some additional level to this array?

File: src/Sylius/Bundle/CoreBundle/Collector/DebugCartCollector.php [hunk 2]
```diff
@@ -0,0 +1,132 @@
...
+    public function collect(Request $request, Response $response, \Exception $exception = null): void
+    {
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function reset(): void
+    {
+        $this->data['cart_id'] = null;
```
Comment (lchrusciel) [inline src/Sylius/Bundle/CoreBundle/Collector/DebugCartCollector.php]: I suppose nulling this field only is not enough

**Comment by gabiudrescu** [comment]
I wouldn't leave this gem to get lost

**Comment by gabiudrescu** [comment]
I would add to this debug panel for cart the promotions that were evaluated for eligibility and why they were rejected or not. With the current implementation, it takes a lot of concentration to understand why a promotion was applied to cart or why it didn't.

and if you have a couple of hundred of them... things can get really really messy.

**Comment by pamil** [comment]
Added an icon for the cart:

CleanShot 2020-05-20 at 19 37 28@2x
CleanShot 2020-05-20 at 19 37 17@2x

**Comment by lchrusciel** [comment]
Thanks, Paweł! :tada:

**Pull request merged**